### ISSUE TITLE
Allow a Node to report that it contains itself

### ISSUE THREAD

**Issue by bmomberger-bitovi** [canjs/can-simple-dom#98]
The line https://github.com/canjs/can-simple-dom/blob/3681610092c1becc9d0fe1b34a09bfe3cc7a473b/lib/document/node.js#L167 conflicts with a recent change I had to make to can-dom-mutate's `contains` function which polyfills some functionality for non-current browsers and/or simplified environments.

* IE11's `contains` always returns `false` when `child` is a text node, so now can-dom-mutate checks the parent node instead, if the child is text.  Sometimes the parent of the text node is null, so this line causes some tests to fail.
* Chrome's `contains` returns true if parent and child are the same, so removing the linked line in question would bring the intended function closer to evergreen implementations.

tl;dr I recommend removing the linked line, as it fixes an issue and improves compatibility with other browsers.

**Comment by justinbmeyer** [comment]
Agree. We do as chrome do.
